cmake/HalideGeneratorHelpers.cmake

```diff
@@ -680,3 +668,13 @@ function(_Halide_target_export_single_symbol TARGET SYMBOL)
         GNU_LD "${CMAKE_CURRENT_BINARY_DIR}/${TARGET}.ldscript"
     )
 endfunction()
+
+function(_Halide_gengen_ensure)
+    # Create a Generator that is GenGen.cpp and nothing else; all it can do is generate a runtime.
+    if (NOT TARGET _Halide_gengen)
+        add_executable(_Halide_gengen )
```

Comment:
This is going to cause problems on Xcode, as it doesn't have a source file associated with it. Unfortunately need an empty, dummy source, here.

Reply:
Er, ok, but this is exactly what we were already doing -- I just moved the code around. Were we broken before?

(Side note: wtf, cmake, how is it possible this isn't supported yet)

Reply:
Done